ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/BasicAction.java

```diff
@@ -275,6 +275,8 @@ public final boolean preventCommit ()
             //	    unlockMutex();
         }
 
+        createPreparedLists();
```

Comment:
Does this truly need to be done in arjunaCore or could it be guarded in XTS alone? It is fine if it is an arjunaCore issue and we should code to fix it but I need to see an arjunacore test case that shows this or reasoning it can't be sole change in XTS